R/zchunk_L222.en_transformation.R

```diff
@@ -81,248 +87,540 @@ module_energy_L222.en_transformation_DISABLED <- function(command, ...) {
-    #   ... code that replicates old, incorrect behavior
-    # } else {
-    #   ... new code with a fix
-    # }
-    #
-    #
-    # NOTE: there are 'match' calls in this code. You probably want to use left_join_error_no_match
-    # For more information, see https://github.com/JGCRI/gcamdata/wiki/Name-That-Function
-    # NOTE: This code uses repeat_and_add_vector
-    # This function can be removed; see https://github.com/JGCRI/gcamdata/wiki/Name-That-Function
+
+    # 2. Build tables for CSVs
+    # 2a. Supplysector information
+    # L222.Supplysector_en: Supply sector information for energy transformation sectors
+
+    A22.sector %>%
+      write_to_all_regions(c("region", "supplysector", "output.unit", "input.unit", "price.unit", "logit.year.fillout", "logit.exponent"), GCAM_region_names) ->
+      L222.Supplysector_en
+
+    # 2b. Subsector information
+    # L222.SubsectorLogit_en: Subsector logit exponents of energy transformation sectors
+
+    A22.subsector_logit %>%
+      write_to_all_regions(c("region", "supplysector", "subsector", "logit.year.fillout", "logit.exponent"), GCAM_region_names) ->
+      L222.SubsectorLogit_en
+
+    if (any(!is.na(A22.subsector_shrwt$year))){
+      A22.subsector_shrwt %>%
+        filter(!is.na(year)) %>%
+        write_to_all_regions(c("region", "supplysector", "subsector", "year", "share.weight"), GCAM_region_names) ->
+        L222.SubsectorShrwt_en
+    }
+
+    if (any(!is.na(A22.subsector_shrwt$year.fillout))){
+      A22.subsector_shrwt %>%
+        filter(!is.na(year.fillout)) %>%
+        write_to_all_regions(c("region", "supplysector", "subsector", "year.fillout", "share.weight"), GCAM_region_names) ->
+        L222.SubsectorShrwtFllt_en
+      }
+
+    # L222.SubsectorInterp_en and L222.SubsectorInterpTo_en: Subsector shareweight interpolation of energy transformation sectors
+
+    if (any(is.na(A22.subsector_interp$to.value))){
+    A22.subsector_interp %>%
+      filter(is.na(to.value)) %>%
+      write_to_all_regions(c("region", "supplysector", "subsector", "apply.to", "from.year", "to.year", "interpolation.function"), GCAM_region_names) ->
+      L222.SubsectorInterp_en
+    }
+
+    if (any(!is.na(A22.subsector_interp$to.value))){
+    A22.subsector_interp %>%
+      filter(!is.na(to.value)) %>%
+      write_to_all_regions(c("region", "supplysector", "subsector", "apply.to", "from.year", "to.year", "to.value", "interpolation.function"), GCAM_region_names) ->
+      L222.SubsectorInterpTo_en
+    }
+
+    # 2c. Technology information
+
+    # L222.StubTech_en: Identification of stub technologies of energy transformation
+    # set up filter to keep all non-first gen bio techs in L222.StubTech_en
+    firstgenbio_techs <- c("corn ethanol", "sugarbeet ethanol", "sugar cane ethanol", "biodiesel")
+
+    # create list of regional stub.technologies
+    A22.globaltech_shrwt %>%
+      write_to_all_regions(c("region", "supplysector", "subsector", "technology"), GCAM_region_names) %>%
```

Comment:
Why is there no use of `LEVEL2_DATA_NAMES` in this chunk?

Reply:
The short answer is that I didn't know it existed or what it does? I was replacing all of the obtuse "names_StubTech" from the old system by explicitly calling the columns, e.g., `<write_to_all_regions(c("region", "supplysector", "subsector", "logit.year.fillout", "logit.exponent"), GCAM_region_names)>`, which seemed much clearer to me. Was I actually supposed to be referencing this database?

Reply:
Ah, sorry for the missing information. Yes please, use `LEVEL2_DATA_NAMES`! Occasionally there's something missing, but almost all of the level 2 outputs should have an entry.

Reply:
No worries, the perils of starting this project after everyone else had gone through the training. I've made the changes and I'll commit once I've wrapped up the other changes.

Reply:
@bpbond sorry one more question on this. There are items in LEVEL2_DATA_NAMES where the appropriate list renames columns: e.g. `L222.StubTechProd_refining` would seem to use `LEVEL2_DATA_NAMES[["StubTechProd"]]` but the old data system columns are `"region" "supplysector" "subsector" "stub.technology" "year" "calOutputValue" "year.share.weight" "subs.share.weight" "share.weight"` and using level2 would replace `share.weight.year` with `year.share.weight` and `share.weight` with `tech.share.weight`. Should I be matching the level2 data names and applying some kind of flag to it, manually matching the old data system list, or somehow trying to change the level2 data names? The old data system uses `names_StubTechYr` and then adds a bunch of custom columns.

Reply:
Follow the old data system please; use `LEVEL2_DATA_NAMES[["StubTechYr"]]` and then add columns as necessary.